src/pybind/mgr/volumes/fs/operations/versions/metadata_manager.py

```diff
@@ -70,7 +81,21 @@ def flush(self):
 
         fd = None
         try:
-            fd = self.fs.open(self.config_path, os.O_WRONLY | os.O_CREAT | os.O_TRUNC, self.mode)
+            if (incarnation != b""):
+                if (incarnation != MetadataManager.EMPTY_INCARNATION):
```

Comment:
For a snapshot preserved volume, we would still leave behind a .meta with a volume state that is "empty"

Reply:
Incarnation is not carried over to the metadata file any longer.